src/React/SyntheticEvent.purs

```diff
@@ -23,53 +23,120 @@ module React.SyntheticEvent
   , SyntheticClipboardEvent'
   , SyntheticCompositionEvent'
   , SyntheticKeyboardEvent'
-  , SyntheticFocusEvent'
   , SyntheticMouseEvent'
   , SyntheticTouchEvent'
   , SyntheticTransitionEvent'
   , SyntheticUIEvent'
   , SyntheticWheelEvent'
 
+  , SyntheticAnimationTransitionEvent'
+  , SyntheticKeyboardMouseTouchEvent'
+  , SyntheticFocusMouseEvent'
+
+  , SyntheticEvent_
+
   , NativeEventTarget
   , NativeEvent
   , NativeDataTransfer
   , NativeAbstractView
   , NativeTouchList
 
+  , bubbles
+  , cancelable
+  , currentTarget
+  , defaultPrevented
+  , eventPhase
+  , isTrusted
+  , nativeEvent
   , preventDefault
   , isDefaultPrevented
   , stopPropagation
   , isPropagationStopped
+  , target
+  , timeStamp
+  , type_
   , persist
+  , animationName
+  , clipboardData
+  , data_
+  , relatedTarget
+  , charCode
+  , key
+  , keyCode
+  , locale
+  , location
+  , repeat
+  , which
+  , button
+  , buttons
+  , clientX
+  , clientY
+  , pageX
+  , pageY
+  , screenX
+  , screenY
+  , changedTouches
+  , targetTouches
+  , touches
+  , altKey
+  , ctrlKey
+  , getModifierState
+  , metaKey
+  , shiftKey
+  , propertyName
+  , pseudoElement
+  , elapsedTime
+  , detail
+  , view
+  , deltaMode
+  , deltaX
+  , deltaY
+  , deltaZ
   ) where
 
 import Prelude
 
 import Control.Monad.Eff (Eff)
 
-type SyntheticEvent = Record (SyntheticEvent' ())
+import Data.Symbol (class IsSymbol, SProxy(..), reflectSymbol)
+
+type SyntheticEvent r
```

Comment:
Should these rows be closed? For example, it's not possible to have both a mouse event and an animation event.

Reply:
Yes. These can be closed. Good catch.